ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -657,22 +678,103 @@ public void shouldInsertInto() throws Exception {
   }
 
   @Test
-  public void shouldHandleErrorResponseFromInsertInto() {
+  public void shouldHandlePullQueryTableException() throws Exception {
+    // When: pull query table
+    final String query = "SELECT * from " + TEST_TABLE + " WHERE K='my_key';";
+    final Exception e = assertThrows(
+        ExecutionException.class, // thrown from .get() when the future completes exceptionally
+        () -> client.executeQuery(query).get()
+    );
+
+    // Then
+    assertThat(e.getCause(), instanceOf(KsqlClientException.class));
+    assertThat(e.getCause().getMessage(), containsString("Received 400 response from server"));
+    assertThat(e.getCause().getMessage(), containsString("table isn't queryable"));
+  }
+
+  @Test
+  public void shouldPushQueryTableAfterInsert() throws Exception {
```

Comment:
The existing test `shouldInsertInto()` already issues a push query after an `insertInto()` call. What additional coverage does this test buy us?

Reply:
`shouldInsertInto` insert into a stream `EMPTY_TEST_STREAM `, this one `insertInto` a table.

Reply:
If that's the purpose of this new test, can we rename the test to `shouldInsertIntoTable()` or similar, in order to better capture what the test is for?